test/thermo/PengRobinsonMFTP_Test.cpp

```diff
@@ -0,0 +1,199 @@
+        moleFracs[0] = r;
+        moleFracs[2] = 1-r;
+        test_phase->setMoleFractions(&moleFracs[0]);
+    }
+
+    std::unique_ptr<ThermoPhase> test_phase;
+};
+
+TEST_F(PengRobinsonMFTP_Test, construct_from_cti)
+{
+	PengRobinsonMFTP* peng_robinson_phase = dynamic_cast<PengRobinsonMFTP*>(test_phase.get());
+    EXPECT_TRUE(peng_robinson_phase != NULL);
+}
+
+TEST_F(PengRobinsonMFTP_Test, chem_potentials)
+{
+    test_phase->setState_TP(298.15, 101325.);
+    // Chemical potential should increase with increasing co2 mole fraction:
+    //      mu = mu_0 + RT ln(gamma_k*X_k).
+    // where gamma_k is the activity coefficient.  Run regression test against values calculated using
+    // the model.
+    const double expected_result[9] = {
+		-4.5736179276418978e+008,
+		-4.5733768577357662e+008,
+		-4.5732940577550375e+008,
+		-4.5732203499398655e+008,
+		-4.5731543698577213e+008,
+		-4.5730950087044573e+008,
+		-4.5730413565940493e+008,
+		-4.5729926602445662e+008,
+		-4.5729482909773445e+008
+    };
+
+    double xmin = 0.6;
+    double xmax = 0.9;
+    int numSteps = 9;
+    double dx = (xmax-xmin)/(numSteps-1);
+    vector_fp chemPotentials(7);
+    for(int i=0; i < 9; ++i)
+    {
+        set_r(xmin + i*dx);
+        test_phase->getChemPotentials(&chemPotentials[0]);
+        EXPECT_NEAR(expected_result[i], chemPotentials[0], 1.e-6);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, activityCoeffs)
+{
+    test_phase->setState_TP(298., 1.);
+
+    // Test that mu0 + RT log(activityCoeff * MoleFrac) == mu
+    const double RT = GasConstant * 298.;
+    vector_fp mu0(7);
+    vector_fp activityCoeffs(7);
+    vector_fp chemPotentials(7);
+    double xmin = 0.6;
+    double xmax = 0.9;
+    int numSteps = 9;
+    double dx = (xmax-xmin)/(numSteps-1);
+
+    for(int i=0; i < numSteps; ++i)
+    {
+        const double r = xmin + i*dx;
+        set_r(r);
+        test_phase->getChemPotentials(&chemPotentials[0]);
+        test_phase->getActivityCoefficients(&activityCoeffs[0]);
+        test_phase->getStandardChemPotentials(&mu0[0]);
+        EXPECT_NEAR(chemPotentials[0], mu0[0] + RT*std::log(activityCoeffs[0] * r), 1.e-6);
+        EXPECT_NEAR(chemPotentials[2], mu0[2] + RT*std::log(activityCoeffs[2] * (1-r)), 1.e-6);
+    }
+}
+
+TEST_F(PengRobinsonMFTP_Test, standardConcentrations)
+{
+    EXPECT_DOUBLE_EQ(test_phase->pressure()/(test_phase->temperature()*GasConstant), test_phase->standardConcentration(0));
+    EXPECT_DOUBLE_EQ(test_phase->pressure()/(test_phase->temperature()*GasConstant), test_phase->standardConcentration(1));
+}
+
+TEST_F(PengRobinsonMFTP_Test, activityConcentrations)
+{
+    // Check to make sure activityConcentration_i == standardConcentration_i * gamma_i * X_i
+    vector_fp standardConcs(7);
+    vector_fp activityCoeffs(7);
+    vector_fp activityConcentrations(7);
+    double xmin = 0.6;
+    double xmax = 0.9;
+    int numSteps = 9;
+    double dx = (xmax-xmin)/(numSteps-1);
+
+    for(int i=0; i < 9; ++i)
```

Comment:
@gkogekar was basing this off of the RedlichKwon test file.  Apparently ++i increments i at the beginning of the loop function, rather than at the end of each loop step.  It currently works as set up, but she can change to `i++` (and change the corresponding data file so that it still works), if desired.

Reply:
So does that mean that the first index used is 1 or 0? In any case, I think its more standard to use `i++`, but I could be wrong. I did notice that `RedlichKwong` uses this convention while I was updating it for #653 so I guessed that was what happened here 😄

Reply:
The first index is 1 in these cases. The data file is also updated accordingly.